cmake/CeleritasLibrary.cmake

```diff
@@ -543,123 +572,53 @@ endfunction()
 # the 3 libraries (static, middle, final) libraries needed
 # for a separatable CUDA library
 function(celeritas_target_link_libraries target)
-  if(NOT CELERITAS_USE_CUDA)
+  if(NOT CELERITAS_USE_VecGeom OR NOT CELERITAS_USE_CUDA)
```

Comment:
The new condition is technically incorrect.  Rather than `NOT CELERITAS_USE_VecGeom` it would `NOT depend_on_any_library_using_relocatable_code_including_celeritas_itself`.   For example if unbeknownst to us, the user select to compile celertias without VecGeom but for whatever reason still with relocatable cuda code, this would fail.

Was there any reason beside optimization of the configure step that motivated this change?

Reply:
https://www.youtube.com/watch?v=hou0lU8WMgo

Nope no other reason. Purity of essence 😉

Reply:
This is however in the same spirit as your comments around line 172:
```cmake
  # Whether we need the special code or not is actually dependent on information
  # we don't have ... yet
  # - whether the user request CUDA_SEPARABLE_COMPILATION
  # - whether the library depends on a library with CUDA_SEPARABLE_COMPILATION code.
  # I.e. this should really be done at generation time.
  # So in the meantime we use CELERITAS_USE_VecGeom as a proxy.

  if(NOT CELERITAS_USE_VecGeom OR NOT CELERITAS_USE_CUDA OR NOT _cuda_sources)
```

Reply:
OK, that's good to know—but our CI and some of our systems are still using 1.1.17 so we should hold off on this change until we require VecGeom 1.2.0. I'll add a note to #421 since requiring 1.2 necessarily requires C++17 so we might as well change at the same time.

Reply:
humm .. the "improvement" were are talking about is current in `celeritas/cmake/FindVecGeom.cmake` , i.e. it should work independently of the VecGeom version used ;)

Reply:
Oh, I see: I thought you were talking about detecting the native `CUDA_SEPARABLE_COMPILATION` CMake property on upstream libraries? Also keep in mind it's impossible to tell in advance (via a configuration-time script that creates the library) what libraries it will be linked against... so most applications will have to use a heuristic such as `USE_CUDA AND USE_VecGeom`.

Reply:
I'm afraid I'm lost now... but this sounds like it's moving outside the scope of this pull request?

Reply:
What I am saying is that this PR should not be adding 'new' `NOT CELERITAS_USE_VecGeom OR ` and (indeed in a future PR) we could probably remove more of them.